### PR TITLE
Week 3 Hand-in

### PR DESCRIPTION
**Pull request by michaelchangdk**

### CODE REVIEW

File: code/index.html [hunk 1]
```diff
@@ -1,30 +1,327 @@
...
+
+  <!-- Header Section -->
+  <header class="header">
+
+    <!-- Navbar / Menu -->
+    <div class="menu">
+      <img class="menu__logo menu__logo--hidden" src="images/hacking.png" alt="brand logo" />
+      <h1 class="brand menu__brand">
+        Brand Name
+      </h1>
```
Comment (michaelchangdk) [inline code/index.html:32]: I forgot to delete this...

File: README.md [hunk 2]
```diff
@@ -1,12 +1,24 @@
...
+
+After that I built from mobile -> desktop. I setup the html structure for the page first with placeholder content. After that I attempted to use BEM for CSS class naming.
+
+After I added all the classes, I started styling in CSS. Once I was satisfied with the mobile version styling, I worked on tablet and desktop media queries. I did a lot of googling to find solutions to styling the radio buttons and checkboxes.
+
+Lastly, I moved onto the JavaScript portion. I deleted the provided code and instead used a W3Schools tutorial on accordions.
+
+I really wanted an action to happen when you pushed the Submit button, so I worked on that in JS as well.
+
+At the end, I discovered while testing that there was a problem when flipping between portrait and landscape on mobile - depending on which orientation you first loaded the page in, the hero image vh would remain the same. I spent quite a lot of time trying to create a javascript solution to this.
```
Comment (Josse79) [inline README.md:17]: Heroic to get this comprehensive code to be perfect even in landscape.

File: code/index.html [hunk 3]
```diff
@@ -1,30 +1,324 @@
...
-      <h1>FAQ</h1>
-
-      <div class="question" id="section1">Section 1</div>
-      <div class="answer">
-        <p>Lorem ipsum...</p>
+
+<head>
+  <meta charset="utf-8" />
+  <!-- Viewport fit: cover is for iphone overflow -->
+  <meta name="viewport" content="width=device-width, initial-scale=1.0, viewport-fit=cover" />
```
Comment (Josse79) [inline code/index.html:7]: I had no idea about viewport-fit=cover, will keep that in mind - very well spotted.

File: code/index.html [hunk 4]
```diff
@@ -1,30 +1,324 @@
...
+        FREQUENTLY ASKED QUESTIONS
+      </h1>
+
+      <button class="subtitle faq__question" id="question1">
+        How does this work?
+      </button>
+      <div class="faq__answer">
+        <p class="text faq__text">
+          Fill out your victim's information in the <strong>R.I.P. INBOX</strong> form. After selecting the various options that might "interest" them, our nifty bot signs them up for newsletters and mailing lists.
+          <br><br>Click <strong>SUBMIT</strong>. Violence achieved.
```
Comment (Josse79) [inline code/index.html:73]: In below "Common Pitfall" I read we shoud avoid <br> and stick to <p> as default display: block, but Im often tempted to use it, and think its difficult living withouth it:-)
(https://www.notion.so/Week-2-common-pitfalls-bbe695289d154ff98d6a247cef873f07)

File: code/index.html [hunk 5]
```diff
@@ -1,30 +1,324 @@
...
+        </label>
+        <label for="phone">
+          <input name="phone" id="phone" type="tel" placeholder="Their phone number (Optional)" />
+        </label>
+
+        <!-- Radio Buttons -->
+        <!-- Email Only Button -->
+        <div class="select-container">
+          <div class="select-container__object">
+            <label for="emailOnly" class="radio--remove">
```
Comment (Josse79) [inline code/index.html:153]: I can see you name your classes with double dash -- , hmmm I wonder if its for a special reason which is in favor of the code...?

File: code/index.html [hunk 6]
```diff
@@ -1,30 +1,324 @@
...
+            </span>
+            </label>
+          </div>
+        </div>
+        <!-- Email and Telemarketing Button -->
+        <div class="select-container">
+          <div class="select-container__object">
+            <label for="emailTelemarketing" class="radio--remove">
+              <input type="radio" id="emailTelemarketing" name="mailOrTele" value="Email and Telemarketing" />
+              <span class="radio--custom"></span>
```
Comment (Josse79) [inline code/index.html:169]: This part of remove and custom the radio buttons seems a bit complicated.
Im impressed you even started that "project".

File: code/index.html [hunk 7]
```diff
@@ -1,30 +1,324 @@
...
+        <div class="checkbox-container">
+          <div class="select-container">
+            <div class="select-container__object">
+              <label for="farming" class="checkbox--remove">
+                <input type="checkbox" id="farming" name="categories" value="farming" />
+                <span class="checkbox--custom"></span>
+            </div>
+            <div class="select-container__text">
+              <span class="signup__options">
+                Farming (50+)
```
Comment (Josse79) [inline code/index.html:195]: Here you can link my website :-)

File: code/index.html [hunk 8]
```diff
@@ -1,30 +1,324 @@
...
+  <div class="black-screen">
+    <div class="black-screen__container">
+      <img class="black-screen__icon" src="images/cementery.png" alt="Cemetary Icon" />
+      <h1 class="title title--large black-screen__text" id="restInPeace">
+        Rest In Peace
+      </h1>
+      <button class="black-screen__button" onclick="history.go(0);">
+        Kill Again?
+      </button>
+    </div>
```
Comment (Josse79) [inline code/index.html:316]: This part is just amazing!

File: code/style.css [hunk 9]
```diff
@@ -1,7 +1,758 @@
-/* the + sign is used to select the sibling that comes immediately next to the element you selected.
-in this case it is the div with the answer class next to the div with the active class.
-this can be used for the FAQ-accordion. read more here: https://www.w3schools.com/css/css_combinators.asp */
+/* Remove Existing Styling (Stolen from Karin) */
```
Comment (Josse79) [inline code/style.css:1]: I did also steel that part from her :-)

File: code/style.css [hunk 10]
```diff
@@ -1,7 +1,758 @@
...
+  font-family: 'Roboto Mono', monospace;
+  font-weight: 700;
+  font-size: 94px;
+  text-align: left;
+  color: #0b0b0b;
+}
+
+/* Page */
+.page {
+  display: grid;
```
Comment (Josse79) [inline code/style.css:73]: Impressive that you used both flex and grid in this css.

File: code/style.css [hunk 11]
```diff
@@ -1,7 +1,758 @@
...
+}
+
+.active:after {
+  content: "\2212";
+  color: #DCD8DC;
+} */
+
+/* Emoji Option */
+.faq__question:after {
+  content: "\1F607";
```
Comment (Josse79) [inline code/style.css:189]: Aha, so this is the code for the emojis.

File: code/style.css [hunk 12]
```diff
@@ -1,7 +1,758 @@
...
+/* Placeholder Text Styling */
+::placeholder {
+  color: #2D2E2F;
+  font-family: 'Montserrat', sans-serif;
+  font-weight: 400;
+  font-size: 14px;
+}
+
+/* Input Fields Styling */
+input[type=text], input[type=tel], input[type=email] {
```
Comment (Josse79) [inline code/style.css:239]: I have to read up on this selectors with the type included. Nice to know its possible without id or class.

File: code/style.css [hunk 13]
```diff
@@ -1,7 +1,758 @@
...
+  color: #dcd8dc;
+  font-family: 'Montserrat', sans-serif;
+  font-weight: 400;
+  font-size: 16px;
+}
+
+
+/* Radio Buttons */
+/* Remove Styling from Radio Buttons */
+.radio--remove {
```
Comment (Josse79) [inline code/style.css:262]: These parts are just awesome. So much code for that small amount of customization. A gold medal for your big effort on this one.

File: code/style.css [hunk 14]
```diff
@@ -1,7 +1,758 @@
...
+
+  .black-screen__icon {
+    max-width: 450px;
+    max-height: 450px;
+  } 
+}
+
+/* Media Query Mobile Landscape Mode */
+
+@media (max-width: 930px) and (orientation: landscape) {
```
Comment (Josse79) [inline code/style.css:673]: This I have to keep in mind, how to select landscape. Great job.

File: code/style.css [hunk 15]
```diff
@@ -1,7 +1,758 @@
...
+  .header {
+    max-height: 100vh;
+    display: grid;
+    grid-template-columns: 1fr 1fr;
+    grid-template-rows: 1fr 1fr;
+  }
+
+
+
+/* Env safe area inset left and right for safari ios when viewport-fit=cover. */
```
Comment (Josse79) [inline code/style.css:686]: How did you find all this? Looks a bit tricky  - but I guess you succeeded. No wonder you were a bit frustrated over Safari....

**Review by Josse79 — COMMENTED** [review]
No doubt that this is a code that is subject for black level.

**Comment by Josse79** [comment]
Love your wording in the commits 👍  :-)

File: code/index.html (hunk 5 above)
Comment (michaelchangdk) [reply code/index.html:153]: I was trying to use BEM naming conventions here! They suggested it somewhere in the week's material but I can't find where now - luckily I bookmarked it: http://getbem.com/introduction/

File: code/index.html (hunk 4 above)
Comment (michaelchangdk) [reply code/index.html:73]: Ooh I should read up on that again. But I agree - `display:block` is a little hard to live without 😭

File: code/index.html (hunk 3 above)
Comment (michaelchangdk) [reply code/index.html:7]: Thank you! It took a lot of googling. When I tested on my cellphone, I noticed that the sides cut off in landscape and I wanted it to fit the entire width of the screen!

File: code/index.html (hunk 6 above)
Comment (michaelchangdk) [reply code/index.html:169]: It was honestly horrifying! A real struggle 😨

File: code/index.html (hunk 8 above)
Comment (michaelchangdk) [reply code/index.html:316]: If you want me to go through it with you and explain what's going on at some point let me know!

I love your daughter's comment 😆

**Pull request closed without merging**